middleware.js

```diff
@@ -1,4 +1,38 @@
 var http = require('http');
 
 exports.createServer = function () {
-};
+  var middlewares = [];
+
+  function next(req, res, index) {
+    if(index == undefined) {
+      index = -1;
+    }
+    return function () {
+      for(var i = index + 1; i < middlewares.length; i++) {
```

Comment:
Also @yorkie how about test cases like

```js
app.use('/goodbye', function (req, res) {
    res.end('goodbye');
});

app.use(function (req, res) {
    res.end('end');
});
```

And a request to `/goodbye` should get `goodbye` instead of `goodbyeend` here, am I right?

Thanks.